### ISSUE TITLE
Tesseract can't detect hyphen

### ISSUE THREAD

**Issue by SpaceWhite** [tesseract-ocr/tesseract#1055]
### Environment

* Tesseract Version: 3.04.01
* Platform: Ubuntu 16.04 LTS 64-bit

### Image
![convert_image](https://user-images.githubusercontent.com/15065085/28657279-4beed71a-72e1-11e7-8764-40bc59ba25fb.png)

### Current Behavior:
055775475566

### Expected Behavior:
0557-54-5566
### Suggested Fix:
Detected number 7 rather than hyphen "-"
Better performance on number and symbol

**Comment by chrismamo1** [comment]
Are you able to do any preprocessing on images in your application? I've found that Tesseract tends to fail on smaller images, seemingly for no reason other than size. For instance, I managed to get Tesseract to properly recognize your image by simply using ImageMagick to scale it up:

    convert original.png -resize 1024x1024 resized.png

**Comment by amitdo** [comment]
https://github.com/tesseract-ocr/tesseract/wiki/ImproveQuality

**Comment by SpaceWhite** [comment]
Thank you for your responses. It's really helpful :)

**Comment by rochus-keller** [comment]
Dear developpers, I can give you a good reason to open the issue again.

I thumbled over the same issue as reported above and was able to narrow it down and find a work-around. The resolution is definitely not the problem. Here is an example: 
![sample.png](https://user-images.githubusercontent.com/2523113/44065227-bfe85f0a-9f69-11e8-997f-1e7e5bc53254.png). As you can see the x height has more than enough pixels and still all hyphens get lost. If I comment out the line "bl_block->PrepareForSplineFitting(page_tr, remove_noise);" in BaselineDetect::ComputeBaselineSplinesAndXheights (file tesseract/textord/baselinedetect.cpp) the hyphens are properly delivered. In version 3.05.02 the mentioned line is only executed if enable_splines is true; the issue also occurs there unless I set enable_splines=false at the top of the method. I didn't analyze yet why PrepareForSplineFitting eats hyphens but will do that if I have time; for now I have a viable work-around and am happy with it. Tesseract is great!

Best R.

**Comment by Shreeshrii** [comment]
seems to work fine with 4.0.0-beta

```
ubuntu@tesseract-ocr:~/tesseract/test$ tesseract hyphen.png - --tessdata-dir ../../tessdata_best
0067-9867-1199-0066
ubuntu@tesseract-ocr:~/tesseract/test$ tesseract hyphen.png - --tessdata-dir ../../tessdata
0067-9867-1199-0066
ubuntu@tesseract-ocr:~/tesseract/test$ tesseract -v
tesseract 4.0.0-beta.4-37-g6295
 leptonica-1.76.0
  libgif 5.1.4 : libjpeg 8d (libjpeg-turbo 1.4.2) : libpng 1.2.54 : libtiff 4.0.6 : zlib 1.2.8 : libwebp 0.4.4 : libopenjp2 2.3.0
ubuntu@tesseract-ocr:~/tesseract/test$
```

**Comment by rochus-keller** [comment]
Thanks for the test. Since it is beta and we cannot afford to upgrade to 4.x until its reasonably stable it would be very welcome if it would also work in the 3.05.x series. And we're using tess in embedded mode without cube, not stand-allone. Btw it also works when I try with the stand-allone 3.03 version.

**Comment by stweil** [comment]
4.0 "beta" does not imply that it is less stable than 3.05. There is currently no one who actively cares for 3.05, for example by backporting all fixes and improvements from 4.0 beta to that older version.

**Comment by amitdo** [comment]
>There is currently no one who actively cares for 3.05, for example by backporting all fixes and improvements from 4.0 beta to that older version.

The easiest way to have new legacy-only release is to make a new 3.06/3.6.0 branch, and change `tesseractmain.cpp` to only accept oem 0 and 1.
